rewatch/src/build/build_types.rs

```diff
@@ -101,6 +104,21 @@ pub struct BuildState {
     pub deps_initialized: bool,
 }
 
+/// Extended build state that includes command-line specific overrides.
+/// Wraps `BuildState` and adds command-specific data like warning overrides.
+/// Used by commands that need to respect CLI flags (e.g., `build`, `watch`).
+///
+/// The separation exists because:
+/// - `clean` command only needs core build data, no CLI overrides
+/// - `build`/`watch` commands need both core data AND CLI overrides
+/// - This prevents the "code smell" of optional fields that are None for some commands
+#[derive(Debug)]
+pub struct BuildCommandState {
```

Comment:
I didn't want to populate BuildState with something that isn't used all the time. Clean doesn't have this for example.

Reply:
I wouldn't mind adding it to the build state, but this is fine!